src/sentry/auth/access.py

```diff
@@ -993,6 +993,7 @@ def from_api_member(
 def from_auth(auth: ApiKey | SystemToken, organization: Organization) -> Access:
     if is_system_auth(auth):
         return SystemAccess()
+    assert not isinstance(auth, SystemToken)
```

Comment:
Are you sure this assert statement will never fail? Or is the AssertionError handled somewhere else?

Reply:
`is_system_auth` is implemented as `isinstance(auth, SystemToken)` -- the type checker would understand this narrowing except `ApiKey` is currently `Any`